updater/lib/dependabot/updater/dependency_group_change_batch.rb

```diff
@@ -92,7 +100,34 @@ def merge_file_to_batch(file, batch)
                          0
                        end
 
-        batch[file.path] = { file: file, changed: true, changes: change_count + 1 }
+        # We explicitly keep updated_dependencies as an empty list if the :dependency_has_directory ff is not enabled
+        batch[file.path] = { file: file, updated_dependencies: [], changed: true, changes: change_count + 1 }
+      end
+
+      def merge_file_and_dependency_changes(updated_dependencies, updated_dependency_files)
+        updated_dependency_files.each do |updated_file|
+          if updated_file.vendored_file?
+            merge_file_and_dependency_changes_to_batch(updated_file, @vendored_dependency_batch, updated_dependencies)
+          else
+            merge_file_and_dependency_changes_to_batch(updated_file, @dependency_file_batch, updated_dependencies)
+          end
+        end
+      end
+
+      def merge_file_and_dependency_changes_to_batch(file, batch, updated_dependencies)
+        change_count = if (existing_file = batch[file.path])
+                         existing_file.fetch(:change_count, 0)
+                       else
+                         # The file is newly encountered
+                         Dependabot.logger.debug("File #{file.operation}d: '#{file.path}'")
+                         0
+                       end
+
+        previous_updated_dependencies = batch[file.path][updated_dependencies] || []
```

Comment:
🔎 It looks like we need `updated_dependencies` to be a `Symbol`. I'd expect this to (almost) always return `nil`

```suggestion
        previous_updated_dependencies = batch[file.path][:updated_dependencies] || []
```